src/Orleans/Async/Cancellation/ICancellationSourcesExtension.cs

```diff
@@ -0,0 +1,12 @@
+﻿using System.Threading.Tasks;
+using Orleans.Concurrency;
+using Orleans.Runtime;
+
+namespace Orleans.Async
+{
+    internal interface ICancellationSourcesExtension : IGrainExtension, IGrain
+    {
+        [AlwaysInterleave]
+        Task CancelTokenSource(GrainCancellationToken token);
```

Comment:
So now there's a case when cancel request arrives before the original one and it fails, as cancellation extension is not yet installed (it's being installed when GCT arrives). Probably the design with extension installing by passing whole GCT instead of Id isn' the most straightforward one; an alternative would be either retrying in case of this fail or passing some `CancellationFailed` exception to user to be retried by him, but the latter case add seemingly unnecessary entity and additional complexity.